core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -548,4 +549,31 @@ class DeleteTopicTest extends QuorumTestHarness {
     TestUtils.waitUntilTrue(() => brokers.exists(_.asInstanceOf[KafkaServer].kafkaController.isActive), "No controller is elected")
     TestUtils.verifyTopicDeletion(zkClient, topic, 2, brokers)
   }
+
+  private def increasePartitions[B <: KafkaBroker](admin: Admin,
+                                           topic: String,
+                                           totalPartitionCount: Int,
+                                           brokersToValidate: Seq[B]
+                                          ): Unit = {
+
+    try {
+      val newPartitionSet: Map[String, NewPartitions] = Map.apply(topic -> NewPartitions.increaseTo(totalPartitionCount))
+      admin.createPartitions(newPartitionSet.asJava)
+    } catch {
+      case e: ExecutionException =>
+        throw e
+    }
+
+    if (brokersToValidate.nonEmpty) {
+      // wait until we've propagated all partitions metadata to all brokers
+      val allPartitionsMetadata = waitForAllPartitionsMetadata(brokersToValidate, topic, totalPartitionCount)
+
+      (0 until totalPartitionCount - 1).map { i =>
+        i -> allPartitionsMetadata.get(new TopicPartition(topic, i)).map(_.leader()).getOrElse(
+          throw new IllegalStateException(s"Cannot get the partition leader for topic: $topic, partition: $i in server metadata cache"))
```

Comment:
this method return nothing, so we don't need to collect the metadata. However, it needs to check the existence of topic partition